### PR TITLE
Remove some internal TimeZone method calls

### PR DESCRIPTION
**Pull request by ptomato**
Found this bug while investigating #1294.

### CODE REVIEW

**Review by cjtenny — APPROVED** [review]
Looks good. I'll collect observability & ordering of this into the test262 work items issue as well!

**Comment by ptomato** [comment]
I went to fix the test262 suite which I had forgotten to do on Friday, and realized that I'd overlooked changing GetOffsetStringFor into BuiltinTimeZoneGetOffsetStringFor. @cjtenny would you mind taking another quick look?

File: polyfill/test/ZonedDateTime/prototype/toPlainDateTime/plain-custom-timezone.js [hunk 1]
```diff
@@ -8,15 +8,14 @@ includes: [compareArray.js]
...
 ];
 
-const dateTime = Temporal.PlainDateTime.from("1963-07-02T12:00:00.987654321");
 const timeZone = new Proxy({
-  getPlainDateTimeFor() {
-    actual.push("call timeZone.getPlainDateTimeFor");
-    return dateTime;
+  getOffsetNanosecondsFor() {
+    actual.push("call timeZone.getOffsetNanosecondsFor");
+    return -8735135802468;
```
Comment (cjtenny) [inline polyfill/test/ZonedDateTime/prototype/toPlainDateTime/plain-custom-timezone.js:18]: are these numbers just different keyboard mashings?

**Review by cjtenny — COMMENTED** [review]
lgtm!

File: polyfill/test/ZonedDateTime/prototype/toPlainDateTime/plain-custom-timezone.js (hunk 1 above)
Comment (ptomato) [reply polyfill/test/ZonedDateTime/prototype/toPlainDateTime/plain-custom-timezone.js:18]: I don't know where `160583136123456789n` came from below, but this is the number that yields a PlainDateTime whose time is 12:00:00.987654321.

**Pull request merged**